presto-bytecode/src/main/java/com/facebook/presto/bytecode/ClassDefinition.java

```diff
@@ -266,6 +270,13 @@ public MethodDefinition declareMethod(
             ParameterizedType returnType,
             Iterable<Parameter> parameters)
     {
+        int parameterCount = Iterables.size(parameters);
+        checkArgument(
+                parameterCount > MAX_ALLOWED_VARARG_ARGUMENTS,
+                "Number of arguments (%d) for vararg functions exceeds limit (%d)",
```

Comment:
`declareMethod` declares a method where `parameters` are plain, formal parameters of the method. I don't think they have something to do with "varargs" (e.g `String... args`), does it?